trustmanager/pkcs11/common/hardwareprivatekey.go

```diff
@@ -0,0 +1,97 @@
+//+build pkcs11
+
+package common
+
+import (
+	"crypto"
+	"crypto/x509"
+	"fmt"
+	"io"
+
+	"github.com/theupdateframework/notary"
+	"github.com/theupdateframework/notary/tuf/data"
+	"github.com/theupdateframework/notary/tuf/signed"
+)
+
+const (
+	// SigAttempts defines maximum attempts to sign an image before aborting
+	SigAttempts = 5
+)
+
+// HardwarePrivateKey represents a private key inside of a Hardwarestore
+type HardwarePrivateKey struct {
+	data.ECDSAPublicKey
+	passRetriever notary.PassRetriever
+	slot          HardwareSlot
+	libLoader     Pkcs11LibLoader
+}
+
+// hardwareSigner wraps a HardwarePrivateKey and implements the crypto.Signer interface
+type hardwareSigner struct {
```

Comment:
What's the need for the extra wrapper? `crypto.Signer` is only implemented by virtue of the embedded `HardwarePrivateKey`?

Reply:
I actualy don't know in that case. Took this from the original yubikey implementation. Seems to be implemented like this since the first yubikey enhanced release. So I don't know what was the intention to do it like that. Didn't want to touch things like this, to keep this restructuring as simple as possible.